### ISSUE TITLE
int / tuple comparisons should be rejected

### ISSUE THREAD

**Issue by josharian** [google/starlark-rust#54]
```
$ starlark-go -c '0 < ()'
Traceback (most recent call last):
  cmdline:1: in <toplevel>
Error: int < tuple not implemented
(exit 1)

$ starlark-rust -c '0 < ()'
True
(exit 0)
```

**Comment by damienmg** [comment]
Starlark in rust used the Python comparison, according to https://github.com/bazelbuild/starlark/blob/master/spec.md#comparisons, it should forbid subtypes ordered comparisons

**Comment by laurentlb** [comment]
fyi, Python 2 returns False, but Python 3 throws an exception. This is why we decided in Starlark to throw an exception.

**Comment by stepancheg** [comment]
No, fix #54, we need to separate `equals` and `compare` operations, which is on my TODO list.

If we simply prohibit `compare` between different types now, `0 == ()` will fail, while it should return `False`.
